source/elements/oneMKL/source/domains/dft/config_params/data_layouts.rst

```diff
@@ -0,0 +1,299 @@
+the :ref:`elementary range of indices considered by
+oneMKL<onemkl_dft_elementary_range_of_indices>`.
+
+Additionally, for in-place transforms (configuration value
+``config_value::INPLACE`` associated with configuration parameter
+``config_param::PLACEMENT``), the smallest stride value must be associated with
+the same dimension in forward and backward domains and the data layouts must
+abide by following "*consistency requirement*": the memory address(es) of
+leading entry(ies) along the last dimension must be identical in forward and
+backward domains. Specifically, considering any :math:`(d+1)`-tuple
+:math:`(m, k_{1}, k_{2}, \dots, k_{d-1}, 0)` within :ref:`valid
+range<onemkl_dft_elementary_range_of_indices>`, the memory address of the
+element of corresponding index value :eq:`eq_idx_data_layout` in forward domain
+(considering the :ref:`implicitly assumed
+type<onemkl_dft_config_data_implicitly_assumed_elementary_data_type>` in forward
+domain) must be identical to the memory address of the element of corresponding
+index value :eq:`eq_idx_data_layout` in backward domain (considering the
+:ref:`implicitly assumed
+type<onemkl_dft_config_data_implicitly_assumed_elementary_data_type>` in
+backward domain). Equivalently,
+
+- for complex descriptors, the offset, stride(s) (and distances, if relevant)
+  must be equal in forward and backward domain;
+- for real descriptors, offsets and strides must satisfy
+  :math:`s^{\text{fwd}}_{j} = 2 s^{\text{bwd}}_{j}\ \forall j \in \lbrace 0,
+  \ldots, d - 1\rbrace` (note that :math:`0 \leq j < d`) and distances, if
+  relevant, must satisfy :math:`l^{\text{fwd}} = 2 l^{\text{bwd}}`. Note that
+  this leads to some data padding being required in forward domain if unit
+  strides are used along the last dimension in forward and backward domains.
+
+.. _onemkl_dft_io_strides_deprecated:
+
+.. rubric:: Configuring strides for input and output data [deprecated, **not** recommended]
+
+Instead of specifying strides by domain, one may choose to specify the strides
+for input and output data sequences. Let
+:math:`s^{\text{x}}_{j}, \ j \in \lbrace 0, 1, \ldots, d\rbrace` be the stride
+values for input (resp. output) data sequences if :math:`\text{x} = \text{i}`
+(resp. :math:`\text{x} = \text{o}`). Such
+:math:`s^{\text{x}}_0, s^{\text{x}}_1, \dots, s^{\text{x}}_d` values may be
+communicated as elements, in that order, of a :math:`(d+1)`-long
+``std::vector<std::int64_t>`` configuration value, set for the (deprecated)
+configuration parameter ``config_param::INPUT_STRIDES`` if
+:math:`\text{x} = \text{i}` (resp. ``config_param::OUTPUT_STRIDES`` if
+:math:`\text{x} = \text{o}`).
+
+The values of :math:`s^{\text{i}}_{j}` and :math:`s^{\text{o}}_{j}` are to be
+used and considered by oneMKL if and only if
+:math:`s^{\text{fwd}}_{j} = s^{\text{bwd}}_{j} = 0, \forall j \in \lbrace 0, 1, \ldots, d\rbrace`.
+In such a case, :ref:`descriptor<onemkl_dft_descriptor>` objects must consider
```

Comment:
```suggestion
The values of :math:`s^{\text{i}}_{j}` and :math:`s^{\text{o}}_{j}` are to be
used and considered by oneMKL if and only if
:math:`s^{\text{fwd}}_{j} = s^{\text{bwd}}_{j} = 0, \forall j \in \lbrace 0, 1, \ldots, d\rbrace`.
(This will happen automatically if ``config_param::INPUT_STRIDES`` and ``config_param::OUTPUT_STRIDES``
are set and ``config_param::FWD_STRIDES`` and ``config_param::BWD_STRIDES`` are not. See note below.)
In such a case, :ref:`descriptor<onemkl_dft_descriptor>` objects must consider
```
I know the note's not far away, but I still think it's nice to explicitly reference it here.